asllib/doc/ASLSemanticsReference.tex

```diff
@@ -19,160 +283,1634 @@
+Arithmetic operators over integer values:
 \begin{mathpar}
-  E\globals \triangleq G
+  \inferrule[add\_int]{ n = a + b }{
+    \binop(\texttt{PLUS}, \nvint(a), \nvint(b)) \evalarrow \nvint(n)
+  }
   \and
-  % E^\text{F} \triangleq F
-  % \and
-  E\locals \triangleq L
+  \inferrule[sub\_int]{ n = a - b }{
+    \binop(\texttt{MINUS}, \nvint(a), \nvint(b)) \evalarrow \nvint(\texttt{n})
+  }
   \and
-  \text{Env}(G, L) \triangleq E
+  \inferrule[mul\_int]{ n = a \times b }{
+    \binop(\texttt{MUL}, \nvint(a), \nvint(b)) \evalarrow \nvint(\texttt{n})
+  }
   \and
-  E[x \mapsto v] \triangleq
-    \left\{
-      \begin{array}{ll}
-        (G[x \mapsto v], L) &\text{if}\ x \in \dom (G) \\
-        (G, L[x \mapsto v]) &\text{otherwise} \\
-      \end{array}
-    \right.
+  \inferrule[div\_int]{  b > 0 \\
+    n = a \div b \\
+    a = n \times b}{
+    \binop(\texttt{DIV}, \nvint(a), \nvint(b)) \evalarrow \nvint(b)
+  }
+  \and
+  \inferrule[div\_int\_error\_denominator\_negative]{  b \leq 0 }{
+    \binop(\texttt{DIV}, \nvint(a), \nvint(b)) \evalarrow \Error(\ErrorDivDenominatorNegative)
+  }
+  \and
+  \inferrule[div\_int\_error\_indivisible]{  b > 0 \\
+  n = a \div b \\
+  n \not\in \Z
+  }{
+    \binop(\texttt{DIV}, \nvint(a), \nvint(b)) \evalarrow \Error(\texttt{"/ indivisibility error!"})
+  }
+  \and
+  \inferrule[fdiv\_int\_pos]{  b > 0 \\
+  a \geq 0 \\
+  n = \lfloor a \div b \rfloor
+  }{ \binop(\texttt{DIVRM}, \nvint(a), \nvint(b)) \evalarrow \nvint(n) }
+  \and
+  \inferrule[fdiv\_int\_neg]{  b > 0 \\
+  a < 0 \\
+  n = -(\lceil (-a) \div b \rceil)
+  }{ \binop(\texttt{DIVRM}, \nvint(a), \nvint(b)) \evalarrow \nvint(n) }
+  \and
+  \inferrule[fdiv\_int\_denominator\_negative]{  b \leq 0 }{ \binop(\texttt{DIVRM}, \nvint(a), \nvint(b)) \evalarrow \Error(\texttt{"DIVRM denominator error!"}) }
+  \and
+  \inferrule[frem\_int\_pos]{  b > 0 \\
+    \binop(\texttt{DIVRM}, \nvint(a), \nvint(b)) \evalarrow c \\
+    n = a - (b \times c)
+  }{ \binop(\texttt{MOD}, \nvint(a), \nvint(b)) \evalarrow \nvint(n) }
+  \and
+  \inferrule[frem\_int\_error\_denominator\_not\_positive]{  b \leq 0
+  }{ \binop(\texttt{MOD}, \nvint(a), \nvint(b)) \evalarrow \Error(\texttt{"MOD denominator error!"}) }
 \end{mathpar}
-%
-The notation $E\globals$ refers to the global part of the environment, and
-$E\locals$ to the local variable mappings, and $\text{Env} (G, L)$ is the
-environment formed with the global mappings in $G$ and the local from $L$.
-%
-The notation $E[x \mapsto v]$ is the environment $E$ modified so $x$ is bound
-to $v$.
-%
-Furthermore, $v?$ refers to $v$ or $\bot$, and the usage of $v$ without a
-question mark implies that $v \neq \bot$.
-%
-In particular, the over-writing of a variable in an environment depends on
-whether the variable is a global or local variable.
 
-\section{Evaluation}
-
-Evaluating a program is evaluating its ``main'' subprogram.
-Constructively, evaluating a program requires following its
```

Comment:
Because I defined a different notation for the global/local environments, which matches the notation in the type reference. Also I defined updating functions in general so there's no need to repeat for environments.
v? is never used.